tests/test_common.js

```diff
@@ -453,3 +453,28 @@ test('percentEncode encodes extra reserved characters', t => {
   t.equal(common.percentEncode('foo+(*)!'), 'foo%2B%28%2A%29%21')
   t.done()
 })
+
+test('normalizeClientRequestArgs throws for invalid URL', async t => {
```

Comment:
@gr2m has expressed a preference for testing our helper functions through real use cases in the public API wherever possible, and it seems to pay off well. It also has the advantage of surfacing code that is no longer needed.

Would you be up for expressing these tests in terms of calls to `http`?

Reply:
Yes please 👍